Fix accessions, sequencing_reads, and uploaded_pathogen_genomes

ttung opened the pull request:
### Description
Added a migration to handle the change.

1. There should only be one UploadedPathogenGenome _OR_ SequencingReads per Sample.
2. There can be more than one accession per sample id.

Depends on #110 

### Test plan
ran migration forwards and backwards.

erinstargazer commented on src/py/aspen/database/models/sample.py:
```diff
@@ -14,6 +18,9 @@
 from .usergroup import Group
 
+if TYPE_CHECKING:
+    from .sequences import SequencingReads, UploadedPathogenGenome
```
What happens if `TYPE_CHECKING` is not true when `SequencingReads` is used on line 167?

erinstargazer replied:
I'm guessing it just ignores the entire line where it's used as it looks like it's just a type specification, is that right?

ttung replied:
When you do `from __future__ import annotations`, it ignores type annotations when not type checking.

Commit:
Fix accessions, sequencing_reads, and uploaded_pathogen_genomes

### Description
Added a migration to handle the change.

1. There should only be one UploadedPathogenGenome _OR_ SequencingReads per Sample.
2. There can be more than one accession per sample id.

### Test plan
ran migration forwards and backwards.

shanaxel42 reviewed (approved):
looks good!
